### PR TITLE
Remove assumption in changeOwnershipOfBlockDevices

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Previously there was assumption that all PVCs that
are passed to the function will be block volume mode
PVCs. This assumption is wrong.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/non-root.go [hunk 1]
```diff
@@ -8,21 +8,44 @@ import (
...
+			continue
+		}
+
+		volumeMode, exists := volumeModes[volume.Name]
+		if !exists {
+			return fmt.Errorf("missing volume status for volume %s", volume.Name)
 		}
+
+		if !types.IsPVCBlock(volumeMode) {
+			continue
```
Comment (vladikr) [inline pkg/virt-handler/non-root.go:41]: I would think that here we will need to change ownership of the volume mount path...
I thinking about a filesystem PV, not a block. Do we have a test where we boot a VM with a filesystem volume?

Comment (xpivarc) [reply pkg/virt-handler/non-root.go:41]: Filesystem volumes are handled here https://github.com/kubevirt/kubevirt/blob/main/pkg/host-disk/host-disk.go#L227

File: pkg/virt-handler/non-root.go [hunk 2]
```diff
@@ -74,7 +98,7 @@ func changeOwnershipOfHostDisks(vmiWithAllPVCs *v1.VirtualMachineInstance, res i
 }
 
 func (d *VirtualMachineController) prepareStorage(vmiWithOnlyBlockPVCS, vmiWithAllPVCs *v1.VirtualMachineInstance, res isolation.IsolationResult) error {
```
Comment (dhiller) [inline pkg/virt-handler/non-root.go:100]: Isn't `vmiWithOnlyBlockPVCS` parameter redundant then? And if that is true, could we rename the parameter `vmiWithAllPVCs` to `vmi` if that makes sense?

**Review by dhiller — COMMENTED** [review]
One question inline.

File: pkg/virt-handler/non-root.go (hunk 2 above)
Comment (xpivarc) [reply pkg/virt-handler/non-root.go:100]: Indeed, thanks for catching it!

Comment (xpivarc) [reply pkg/virt-handler/non-root.go]: Fixed

File: pkg/virt-handler/non-root.go (hunk 1 above)
Comment (vladikr) [reply pkg/virt-handler/non-root.go:41]: yep, that's right, thanks.

**Comment by iholder-redhat** [comment]
/lgtm
Thanks!

**Comment by dhiller** [comment]
/hold

will not succeed due to operator lane failures anyway

**Pull request merged**